front/app/api/phases/usePhases.ts

```diff
@@ -4,16 +4,22 @@ import fetcher from 'utils/cl-react-query/fetcher';
 import phasesKeys from './keys';
 import { IPhases, PhasesKeys } from './types';
 
-const fetchPhases = ({ projectId }: { projectId: string }) =>
-  fetcher<IPhases>({
+const fetchPhases = async ({
+  projectId,
+}: {
+  projectId: string | undefined;
+}) => {
+  return fetcher<IPhases>({
     path: `/projects/${projectId}/phases`,
     action: 'get',
   });
+};
```

Comment:
> I'll add it to the FE CDM to hear your thoughts
Yeah sounds good. I agree with you, it's kind of ugly that `projectId` now seems like it can be `undefined` (even though in practice it never is because of the `enabled` option for `react-query`). But I feel like this is a more fundamental problem with react hooks and the fact that you can't call them conditionally.